instrumentation/kafka-clients-metrics-3.0.0/src/main/java/com/nr/instrumentation/kafka/NewRelicMetricsReporter.java

```diff
@@ -34,6 +35,16 @@ public class NewRelicMetricsReporter implements MetricsReporter {
 
     private final Map<String, KafkaMetric> metrics = new ConcurrentHashMap<>();
 
+    private final List<String> nodes;
+
+    public NewRelicMetricsReporter() {
+        this.nodes = Collections.emptyList();
+    }
+
+    public NewRelicMetricsReporter(List<String> nodes) {
+        this.nodes = nodes;
```

Comment:
To prevent the concatenation of Strings every harvest, the metric name for the nodes could be calculated here.

Reply:
I move the creation of the metric names to init, does that make sense to you?